File: programmers/최대공약수와 최소공배수.py
def solution(n, m):
    answer = []
    
    # x = min(n, m)
    # y = max(n, m)
    
    def gdc(n, m):
        while m > 0:
            n, m = m, n % m

        answer.append(n)
    
    gdc(n, m)
    
    lcm = answer[0] * (n // answer[0]) * (m // answer[0])
    answer.append(lcm)
    
    return answer

File: programmers/level1/최대공약수와_최소공배수.cpp
#include <vector>

using namespace std;

int GCD(int a, int b) { // 유클리드 호제법?
    int r;
    while (b != 0) {
        r = a % b;
        a = b;
        b = r; // b로 나누어 떨어졌는가
    }
    return a;
} // ex) 3,12 -> 12,3 -> 3,0

int LCM(int a, int b, int gcd) {
    return a * b / gcd; // 공식
}

vector<int> solution(int n, int m) {
    int gcd = GCD(n,m);
    int lcm = LCM(n,m,gcd);
    
    return { gcd, lcm };
}